docs/content/contributing/GradlePlugins.md

```diff
@@ -0,0 +1,162 @@
+        }
+    }
+}
+
+// This wrapper is needed only for Worker API
+// It can be started in another process. Thus, it has to prepare dependencies for the real work.
+
+abstract class FeatureWorkerAction : WorkAction<Parameters> {
+
+    interface Parameters : WorkParameters {
+        fun getIntegrationApiConfig(): Property<IntegrationApiConfig>
+    }
+
+    override fun execute() {
+        val api = IntegrationApiConfig.Impl(parameters.getIntegrationApiConfig().get())
+        val action = FeatureAction(
+            integrationApi = api
+        )
+        action.execute()
+    }
+}
+
+// This class is responsible for the real work.
+// The less it knows about Gradle, the better.
+
+class FeatureAction(
+    private val integrationApi: IntegrationApi
+) {
+    fun execute() {
+        // Do the real work here
+    }
+}
+
+// Now you can use simple mocks to test the action.
+@Test
+fun test() {
+    val integrationApi = mock<IntegrationApi>()
+    whenever(integrationApi.foo).thenReturn(bar())
+
+    val action = FeatureAction(integrationApi) < --No Gradle abstractions here
+        action.execute()
+
+    assertThat(...)
+}
+```
+
+### Integration tests
+
+Apply a convention plugin:
+
+```kotlin
+plugins {
+    id("convention.gradle-testing")
+}
+```
+
+Place tests in `src/gradleTest/kotlin`
+
+For simple cases you can create dummy instance of Project
+by [ProjectBuilder](https://docs.gradle.org/current/javadoc/org/gradle/testfixtures/ProjectBuilder.html)
+
+```kotlin
+val project = ProjectBuilder.builder().build()
+
+val task = project.tasks.register<TestTask>("testTask") {}
+
+task.get().doStuff()
+```
+
+When you need to run a real build, use [Gradle Test Kit](https://docs.gradle.org/current/userguide/test_kit.html).\
+See ready utilities in `:test-project` module.
+
+#### Debugging
+
+## Debugging
+
+In `GradleTestKit.kt` `fun gradlew()` set `withDebug(true)` to be able to debug gradle plugins. 
+
+Disabled by default, because breaks tests even without debugging when android gradle plugin applied.
+
+See [Gradle issue tracker](https://github.com/gradle/gradle/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+withDebug) about reasons.
+
+### Run tests via CLI
+
+`./gradlew test` - runs unit tests
+`./gradlew gradleTest` - runs gradle integration tests
+
+Add `--continue` if you don't need default fail fast on first failure
+
+To run single test, package or class add `--tests package.class.method`, but keep in mind that it works only for a
+single project
+
+## Best practices
+
+### Feature toggles
+
+The plugin may break and blocks the work of other developers. Making the plugin unpluggable gives you time for a fix.
+
+```kotlin
+open class MyPlugin : Plugin<Project> {
+
+    override fun apply(project: Project) {
+        // In each property use prefix `avito.<plugin>`
+        // It makes it easy to find it in the future
+        if (!project.getBooleanProperty("avito.my_plugin.enabled", default = false)) {
```

Comment:
Let's show reading in a compatible with configuration cache way:

```kotlin
val Project.pluginIsEnabled: Boolean
        get() = providers
            .gradleProperty(enabledProp)
            .forUseAtConfigurationTime()
            .map { it.toBoolean() }
            .getOrElse(true)
```

Also, enabled by default, because disabling is a corner case. 
The plugin should work after applying by default.

Reply:
fixed